server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/CountyReportDownload.java

```diff
@@ -66,9 +98,22 @@ public AuthorizationType requiredAuthorization() {
   // necessary to break out of the lambda expression in case of IOException
   @SuppressWarnings("PMD.ExceptionAsFlowControl")
   public String endpoint(final Request the_request, final Response the_response) {
+    // we know we have either state or county authentication; this will be null
+    // for state authentication
+    County county = Main.authentication().authenticatedCounty(the_request);
+
+    if (county == null) {
+      county = 
+          Persistence.getByID(Long.parseLong(the_request.queryParams(COUNTY)), County.class);
+      if (county == null) {
+        badDataContents(the_response, "county " + the_request.queryParams(COUNTY) +
+                                      " does not exist");
+      }
+      assert county != null; // makes FindBugs happy
```

Comment:
Do we really want to have an assertion failure here, if a DOS user doesn't specify a valid county ID? Will this interrupt the HTTP response?

Reply:
Answer: we will never reach that code, because `badDataContests()` calls `halt()`.

Reply:
badDataContents halts first in any situation where the assertion would fail.